orchestrationSpecs/packages/argo-workflow-builders/src/models/templateBuilder.ts

```diff
@@ -217,17 +217,22 @@ export class TemplateBuilder<
 
     addContainer<
         FirstBuilder extends ContainerBuilder<ContextualScope, InputParamsScope, any, any, any, any>,
-        FinalBuilder extends ContainerBuilder<ContextualScope, InputParamsScope, any, any, any, any>
+        FinalBuilder extends ContainerBuilder<ContextualScope, InputParamsScope,
+        // TODO, figure out how to type check against Container in k8s types
+         GenericScope & { image: any; resources: any; },
```

Comment:
What stops you from replacing `{ image: any; resources: any; }` with Container? Are there other fields that we don't support yet? Would we find all of the missing fields that we expect?